File: authorization-admin-service-spring/src/main/java/eu/rudisch/authorization/admin/web/controller/resource/UserRestRep.java
package eu.rudisch.authorization.admin.web.controller.resource;

import java.util.List;
import java.util.stream.Collectors;

import eu.rudisch.authorization.admin.model.Role;
import eu.rudisch.authorization.admin.model.User;
import lombok.Data;

@Data
public class UserRestRep {

	private int id;
	private String username;
	private String password;
	private String passwordRepeat;
	private String email;
	private boolean enabled;
	private boolean accountExpired;
	private boolean credentialsExpired;
	private boolean accountLocked;
	private List<String> roleNames;

	public static UserRestRep fromUser(User user) {
		UserRestRep userRestRep = new UserRestRep();

		userRestRep.id = user.getId();
		userRestRep.username = user.getUsername();
		userRestRep.password = user.getPassword();
		userRestRep.email = user.getEmail();
		userRestRep.enabled = user.isEnabled();
		userRestRep.accountExpired = !user.isAccountNonExpired();
		userRestRep.accountLocked = !user.isAccountNonLocked();
		userRestRep.credentialsExpired = !user.isCredentialsNonExpired();
		userRestRep.roleNames = user.getRoles().stream()
				.map(Role::getName)
				.collect(Collectors.toList());

		return userRestRep;
	}
}

File: authorization-admin-service-spring-kotlin/src/main/kotlin/eu/rudisch/authorization/admin/web/controller/resource/UserRestRep.kt
package eu.rudisch.authorization.admin.web.controller.resource

import eu.rudisch.authorization.admin.model.User
import java.util.stream.Collectors

data class UserRestRep(val id: Int,
					   val username: String,
					   val password: String,
					   val passwordRepeat: String,
					   val email: String,
					   val enabled: Boolean,
					   val accountExpired: Boolean,
					   val credentialsExpired: Boolean,
					   val accountLocked: Boolean,
					   val roleNames: List<String>) {

	companion object {
		@JvmStatic
		fun fromUser(user: User): UserRestRep {
			return UserRestRep(user.id,
					user.username,
					user.password,
					user.password,
					user.email,
					user.enabled,
					!user.accountNonExpired,
					!user.accountNonLocked,
					!user.credentialsNonExpired,
					user.roles.stream()
							.map { it.name }
							.collect(Collectors.toList<String>()))
		}
	}
}